### PR TITLE
39fixes

### PR DESCRIPTION
**Pull request by tbuschto**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
     "noUnusedLocals": false,
     "jsx": "react",
     "jsxFactory": "JSX.createElement",
+    "typeRoots": [],
```
Comment (cpetrov) [inline tsconfig.json:16]: Why is this necessary?

File: src/api/ActionMapper.ts [hunk 2]
```diff
@@ -26,9 +26,10 @@ export type Callbacks<T extends object> = {
   apply?: MappedRules
 };
 
-type ActionCreator<TargetFunction extends (...args: any[]) => any, Action> = TargetFunction extends Listeners<any>
-  ? (ev: EventOfListeners<TargetFunction>) => Action
-  : (...args: Parameters<TargetFunction>) => Action;
+type ActionCreator<TargetFunction, Action>
```
Comment (cpetrov) [inline src/api/ActionMapper.ts:29]: Why was this necessary? Would you like to mention this in the commit message?

File: tsconfig.json (hunk 1 above)
Comment (tbuschto) [reply tsconfig.json:16]: Exclude parent directories from auomatic type detection

File: src/api/ActionMapper.ts (hunk 2 above)
Comment (tbuschto) [reply src/api/ActionMapper.ts:29]: Honestly not sure, TS4.8 just doesn't seem to like narrow type parameters on mapped types, which where this is used.
